frontend/src/components/__tests__/ErrorBoundary.test.tsx

```diff
@@ -0,0 +1,67 @@
+import {render, waitFor} from '@testing-library/react'
+import ErrorBoundary from '../ErrorBoundary'
+import i18n from 'i18next'
+import {I18nextProvider, initReactI18next} from 'react-i18next'
+
+i18n.use(initReactI18next).init({
+  resources: {},
+  lng: 'en',
+})
+
+const MockProviders = (props: any) => (
+  <I18nextProvider i18n={i18n}>{props.children}</I18nextProvider>
+)
+
+const Child = () => {
+  throw new Error()
+}
+
+describe('Given an ErrorBoundary component', () => {
+  describe('when there is an error in a Child component', () => {
+    it('should render the fallback ui', async () => {
+      const {getByText} = await waitFor(() =>
+        render(
+          <MockProviders>
+            <ErrorBoundary>
+              <Child />
+            </ErrorBoundary>
+          </MockProviders>,
+        ),
+      )
+      expect(getByText('errorBoundary.modal.header')).toBeTruthy()
+    })
+  })
+
+  describe('when the component is mounted', () => {
+    let mockWindowObject: any
+
+    beforeEach(async () => {
+      mockWindowObject = {
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+      }
+
+      render(
+        <MockProviders>
+          <ErrorBoundary windowObject={mockWindowObject}>
+            some content
+          </ErrorBoundary>
+        </MockProviders>,
+      )
+    })
+
+    it('should listen for unhandledrejection and error events', () => {
+      expect(mockWindowObject.addEventListener).toHaveBeenCalledTimes(2)
```

Comment:
I understand your point, that the test doesn't cover properly our use case.

Unfortunately the  `unhandledRejection` handler is not testable anymore, as pointed out in this [GitHub issue](https://github.com/facebook/jest/issues/5620) [](https://github.com/facebook/jest/issues/5620). 

There is a [workaround](https://johann.pardanaud.com/blog/how-to-assert-unhandled-rejection-and-uncaught-exception-with-jest/) provided by a user which involves creating a boostrap file for jest in order not to mock the Node `process` variable. Honestly it looks quite cumbersome for our purpose.

Reply:
One way would be to grab the registered handler and call it yourself in the test. I'd skip this test since we don't care that much about logging on the console, and I'd add it when we'll use the actual logger

Example

```ts
handlers = {}

addEventListener = jest.mock((event, cb)) => {
  handlers[event] = cb
})

// ... later in the test

handlers.unhandledrejection()
expect(mockLogger).toHaveBeenCalled(...)


```